### DISCUSSION TITLE
frame_latlon_and_rates rate type

### DISCUSSION THREAD

**Discussion by mmmvvvppp** [skyfielders/python-skyfield#882] [Q&A] [answered]
I'm reading the documentation at https://rhodesmill.org/skyfield/examples.html#at-what-rate-is-a-target-moving-across-the-sky and it describes that the 4th and 5th values returned from `frame_latlon_and_rates()` are the altitude rate and azimuth rate, respectively.  The code documents the function as:

```    
def frame_latlon_and_rates(self, frame):
        """Return a reference frame longitude, latitude, range, and rates.

        Return a 6-element tuple of 3 coordinates and 3 rates-of-change
        for this position in the given reference ``frame``:

        * Latitude :class:`~skyfield.units.Angle` from +90° north to −90° south
        * Longitude :class:`~skyfield.units.Angle` 0°–360° east
        * Radial :class:`~skyfield.units.Distance`
        * Latitude :class:`~skyfield.units.AngleRate`
        * Longitude :class:`~skyfield.units.AngleRate`
        * Radial :class:`~skyfield.units.Velocity`
        ...
```

To me this implies that the rates are latitude and longitude rates of change, and the variable names also indicate that as well: `d, lat, lon, d_rate, lat_rate, lon_rate = _to_spherical_and_rates(r, v)`

The code I have that exercises this function is very similar to the example and the `frame` argument is a `GeographicPosition`:

```
        ts = load.timescale()
        t = ts.utc(2022, 9, 30, 0, 0)
        geographic_pos = wgs84.latlon(0 * N, 0 * E)
        vec_sum = sat - geographic_pos
        icrf = vec_sum.at(t)
        alt, az, distance, alt_rate, az_rate, range_rate = icrf.frame_latlon_and_rates(
            geographic_pos
        )
```

Questions:
1. Do the results of `frame_latlon_and_rate()s`  include lat/lon rates or alt/az rates when passed a `GeographicPosition` as the argument?  
2. Is the documentation in code correct?

**Comment by brandon-rhodes** [comment]
(Did you perhaps press the submit button by accident before finishing your complete question?)

**Reply by mmmvvvppp** [reply]
Apologies Brandon, I have updated the description with two specific questions.  I can't mentally rectify what appears to me to be conflicting descriptions from examples and in the code, and I assume I'm just uneducated about the terminology.

**Reply by brandon-rhodes** [reply, marked as answer]
1. Given a geographic position and its horizon, "latitude" measured up and down from the horizon is more specifically known as "altitude" or (among satellite folks) as "elevation", and the "longitude" measured around the great circle of the horizon is more specifically "azimuth". So altitude and azimuth will be returned, if I'm reading your code correctly.
2. Yes, the documentation is fine, but maybe a bit brief? I think I had imagined people finding the routine from the main documentation first and seeing an example. I should probably update that docstring to note, for folks who focus on the docstring, that in horizon-based coordinates, the "latitude" is elevation/altitude and the "longitude" is azimuth.

**Reply by mmmvvvppp** [reply]
Thanks so much!  That does clarify things and updating the docstring would be appreciated.  I found the example as you had imagined, but the type hints that pop up with the docstring gave me pause.
